### PR TITLE
Add .i extension for preprocessed C files to known file extensions

### PR DESCRIPTION
**Pull request by cschlote**
Added .i as extension for preprocessed C files, which can be fed directly
to DMD. The file is then processed as C source file.

Supporting .i as file exzension allows to preprocess C code with GCC -E and
store the output to some file for further processing by DMD.

### CODE REVIEW

File: src/dmd/dmodule.d [hunk 1]
```diff
@@ -1042,7 +1047,7 @@ extern (C++) final class Module : Package
 
         /* If it has the extension ".c", it is a "C" file.
          */
-        if (FileName.equalsExt(arg, c_ext))
+        if (FileName.equalsExt(arg, c_ext) || FileName.equalsExt(arg, i_ext))
```
Comment (thewilsonator) [inline src/dmd/dmodule.d:1050]: Please update the comment too.

**Review by thewilsonator — COMMENTED** [review]
Please also add a test case

**Review by WalterBright — APPROVED** [review]

**Comment by WalterBright** [comment]
Needs a changelog entry, too.

File: src/dmd/dmodule.d (hunk 1 above)
Comment (cschlote) [reply src/dmd/dmodule.d:1050]: Comment updated,
Adding a test is WIP. I'm working through the test infrastructure of DMD and try to find the proper place for it. Is there any 'importc' releated test, yet?

**Comment by cschlote** [comment]
Testcase added: ./run.d runnable/importc-test1.sh

**Comment by cschlote** [comment]
There is an open issue with the C compiler to use on Windows and Mac. Faked input '.i' file as workaround.

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**